src/Swarm/Game/Step.hs

```diff
@@ -1873,7 +1875,9 @@ onTarget rid act = do
         Nothing -> return ()
         Just tgt -> do
           tgt' <- execState @Robot tgt act
-          robotMap . ix rid .= tgt'
+          if tgt' ^. selfDestruct
+            then deleteRobot rid
+            else robotMap . ix rid .= tgt'
```

Comment:
@byorgey this took me a while to figure out, but now it's fixed and destroyed robots will be correctly destroyed immediately, instead of hanging around on the map for a while. :slightly_smiling_face: